### PR TITLE
refactor main `Maxwell` class into a reusable entity.

### PR DESCRIPTION
**Pull request by osheroff**
this forces us to exercise all the code paths in the `Maxwell` application in a very real way.

This, I swear, is the last refactor for awhile.

@zendesk/rules

### CODE REVIEW

File: src/main/java/com/zendesk/maxwell/schema/MysqlSchemaStore.java [hunk 1]
```diff
@@ -26,20 +28,22 @@ public MysqlSchemaStore(ConnectionPool maxwellConnectionPool,
 							BinlogPosition initialPosition,
 							CaseSensitivity caseSensitivity,
 							MaxwellFilter filter,
-							boolean replayMode) {
+							boolean readOnly) {
 		super(replicationConnectionPool, caseSensitivity, filter);
+		this.replicationConnectionPool = replicationConnectionPool;
```
Comment (dadah89) [inline src/main/java/com/zendesk/maxwell/schema/MysqlSchemaStore.java]: do you need to assign replicationConnectionPool and filter if you're calling super with them?

File: src/main/java/com/zendesk/maxwell/MaxwellContext.java [hunk 2]
```diff
@@ -76,8 +81,11 @@ public Connection getRawMaxwellConnection() throws SQLException {
 	}
 
 	public void start() {
-		SchemaScavenger s = new SchemaScavenger(this.maxwellConnectionPool);
```
Comment (dadah89) [inline src/main/java/com/zendesk/maxwell/MaxwellContext.java]: we're not going to run the SchemaScavenger anymore?

File: src/main/java/com/zendesk/maxwell/BufferedMaxwell.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+/**
+ * Created by ben on 8/27/16.
+ */
+public class BufferedMaxwell extends Maxwell {
+	public BufferedMaxwell(MaxwellConfig config) throws SQLException {
+		super(config);
+		config.producerType = "buffer";
+	}
+
+	public RowMap getRow(long timeout, TimeUnit unit) throws IOException, InterruptedException {
```
Comment (dadah89) [inline src/main/java/com/zendesk/maxwell/BufferedMaxwell.java]: totally nitpick, the name is a bit weird. At first I thought about MaxwellReplicator#getRow... getProducedMessage/getSentRecord/pollProducerRecords/whatever name that makes sense in english =)

File: src/main/java/com/zendesk/maxwell/MaxwellContext.java (hunk 2 above)
Comment (osheroff) [inline src/main/java/com/zendesk/maxwell/MaxwellContext.java]: nah, we pretty much keep schemas for all time now.  need to tear down the class too I htink.

Comment (dadah89) [inline src/main/java/com/zendesk/maxwell/MaxwellContext.java]: I think so, couldn't find other references to it

File: src/test/java/com/zendesk/maxwell/MaxwellTestSupport.java [hunk 4]
```diff
@@ -89,75 +90,82 @@ public static MaxwellContext buildContext(int port, BinlogPosition p, MaxwellFil
...
-						conn.setCatalog(context.getConfig().databaseName);
-						return conn;
-					}
-				};
-			}
-		};
+		// wait for it to come through
+		maxwell.getRow(5, TimeUnit.SECONDS);
 
-		TestMaxwellReplicator p = new TestMaxwellReplicator(schemaStore, producer, bootstrapper, context, start, endPosition);
```
Comment (dadah89) [inline src/test/java/com/zendesk/maxwell/MaxwellTestSupport.java]: looks like TestMaxwellReplicator isn't being used anymore

**Comment by dadah89** [comment]
lgtm 👍

File: src/main/java/com/zendesk/maxwell/BufferedMaxwell.java (hunk 3 above)
Comment (osheroff) [inline src/main/java/com/zendesk/maxwell/BufferedMaxwell.java]: i renamed this to `poll` in a future pr.  I can backport.

**Pull request merged**